[Profiler] Add pattern to detect if TF32 is available but not used

davidchencsl opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81740
* #81733
* #81501
* __->__ #81273
* #81056

Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

Differential Revision: [D37799414](https://our.internmc.facebook.com/intern/diff/D37799414)

davidchencsl committed:
Add pattern to detecif TF32 is not used

[ghstack-poisoned]

davidchencsl committed:
Update on "[Profiler] Add pattern to detect if TF32 is available but not used"


Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

[ghstack-poisoned]

davidchencsl commented:
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37799414).

robieta commented on test/test_profiler.py:
```diff
@@ -1553,6 +1556,22 @@ def case4():
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
+    def test_profiler_fp32_matmul_pattern(self):
+        has_tf32 = 1
```
It should probably also be a property of `FP32MatMulPattern`. Maybe make `skip` a `@property`?

robieta commented on torch/profiler/_pattern_matcher.py:
```diff
@@ -169,6 +170,35 @@ def match(self, event: _ProfilerEvent):
+        self.description = (
+            "You are currently using GPU that supports TF32. "
+            "Please enable TF32 by setting 'torch.backends.cuda.matmul.allow_tf32 = True'")
+        for arch in torch.cuda.get_arch_list():
```
See comment above

robieta reviewed (commented):
Again, couple of minor nits but looks good overall.

davidchencsl committed:
Update on "[Profiler] Add pattern to detect if TF32 is available but not used"


Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

Differential Revision: [D37799414](https://our.internmc.facebook.com/intern/diff/D37799414)

[ghstack-poisoned]

davidchencsl committed:
Update on "[Profiler] Add pattern to detect if TF32 is available but not used"


Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

Differential Revision: [D37799414](https://our.internmc.facebook.com/intern/diff/D37799414)

[ghstack-poisoned]

davidchencsl commented:
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37799414).

davidchencsl committed:
Update on "[Profiler] Add pattern to detect if TF32 is available but not used"


Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

Differential Revision: [D37799414](https://our.internmc.facebook.com/intern/diff/D37799414)

[ghstack-poisoned]

davidchencsl commented:
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37799414).

davidchencsl committed:
Update on "[Profiler] Add pattern to detect if TF32 is available but not used"


Summary: This commit adds recording of a global flag allow_tf32 in all torch ops. Then I add a pattern that detects if a user is setting tf32 flag or not, recommend he/she to do so if is not set.

Test Plan: Added a test in test_profiler.py

Differential Revision: [D37799414](https://our.internmc.facebook.com/intern/diff/D37799414)

[ghstack-poisoned]

davidchencsl commented:
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37799414).

davidchencsl commented:
@pytorchbot merge
